pkg/generator/generator_test.go

```diff
@@ -0,0 +1,643 @@
+/*
+Copyright 2022 Adevinta
+*/
+
+package generator
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/adevinta/vulcan-agent/jobrunner"
+	"github.com/adevinta/vulcan-local/pkg/config"
+	"github.com/adevinta/vulcan-local/pkg/gitservice"
+	"github.com/adevinta/vulcan-local/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	loggerUser *logrus.Logger
+)
+
+func init() {
+	if len(os.Args) > 1 && os.Args[1][:5] == "-test" {
+		loggerUser = logrus.New()
+		loggerUser.SetFormatter(&logrus.TextFormatter{
+			DisableColors:   false,
+			FullTimestamp:   true,
+			TimestampFormat: time.RFC3339,
+			ForceColors:     true,
+		})
+	}
+}
+
+func errToStr(err error) string {
```

Comment:
Why don't you just use the testutil function?

Reply:
I did it in the same way that it was done in vulcan-api. I have to ask someone what is the reason to do this in a indirect way.

Reply:
makes sense, I would like to know too